libtransact/src/workload/xo.rs

```diff
@@ -0,0 +1,182 @@
+use crate::batch::BatchBuilder;
+use crate::batch::BatchPair;
+use crate::signing::hash::HashSigner;
+use crate::signing::Signer;
+use crate::transaction::HashMethod;
+use crate::transaction::TransactionBuilder;
+use crate::transaction::TransactionPair;
+use crate::workload::error::WorkloadError;
+use crate::workload::BatchWorkload;
+use crate::workload::TransactionWorkload;
+
+use rand::distributions::Alphanumeric;
+use rand::prelude::*;
+use rand_hc::Hc128Rng;
+
+use sha2::{Digest, Sha512};
+
+static FAMILY_NAME: &str = "xo";
+static FAMILY_VERSION: &str = "0.1";
+static NONCE_SIZE: usize = 32;
+
+pub struct XoTransactionWorkload {
+    rng: Hc128Rng,
+    signer: Box<Signer>,
+}
+
+impl XoTransactionWorkload {
+    pub fn new() -> Self {
+        let rng = Hc128Rng::from_entropy();
+        let signer = HashSigner::new();
+
+        XoTransactionWorkload {
+            rng,
+            signer: Box::new(signer),
+        }
+    }
+
+    pub fn new_with_seed(seed: u64) -> XoTransactionWorkload {
+        let rng = Hc128Rng::seed_from_u64(seed);
+        let signer = HashSigner::new();
+
+        XoTransactionWorkload {
+            rng,
+            signer: Box::new(signer),
+        }
+    }
+}
+
+impl TransactionWorkload for XoTransactionWorkload {
+    fn next_transaction(&mut self) -> Result<TransactionPair, WorkloadError> {
+        let nonce = self
+            .rng
+            .sample_iter(&Alphanumeric)
+            .take(NONCE_SIZE)
+            .collect::<String>()
+            .as_bytes()
```

Comment:
These two calls can be replaced with `into_bytes()`

Reply:
Fixed